### PR TITLE
fix SVG selector widget not reacting to changes

### LINKED ISSUE #43802: 3.20.0 regression: layout image item's SVG browser broken
Well, I wish I had created this layout 72 hours prior to 3.20.0 release :wink: the SVG browser in the image item's properties panel isn't working.

*Steps to reproduce*
- Create a new layout
- Add an image item (note: north arrow item is also affected)
- Try to use the SVG browser to change the SVG image
- Notice nothing happens

### PR DESCRIPTION
**Pull request by 3nids**
fixes #43802

test coming shortly

### CODE REVIEW

File: src/gui/layout/qgslayoutpicturewidget.cpp [hunk 1]
```diff
@@ -472,7 +473,7 @@ void QgsLayoutPictureWidget::sourceChanged( const QString &source )
   if ( mPicture )
   {
     mPicture->beginCommand( tr( "Change Picture" ) );
-    mPicture->setPicturePath( source, QgsLayoutItemPicture::FormatSVG );
```
Comment (nyalldawson) [inline src/gui/layout/qgslayoutpicturewidget.cpp:475]: You should be explicitly stating the type here -- why the change?

Comment (3nids) [reply src/gui/layout/qgslayoutpicturewidget.cpp:475]: if you are in Raster mode, the image is not displayed then.

would this be better to you:
    `mPicture->setPicturePath( source, mRadioSVG->isChecked() ? QgsLayoutItemPicture::FormatSVG : QgsLayoutItemPicture::FormatRaster );`

but this whole think svg/raster is really painful, though I don't have any simple proposition to improve the situation.

Comment (nyalldawson) [reply src/gui/layout/qgslayoutpicturewidget.cpp:475]: > would this be better to you:
mPicture->setPicturePath( source, mRadioSVG->isChecked() ? QgsLayoutItemPicture::FormatSVG : QgsLayoutItemPicture::FormatRaster );

Yes please.

> but this whole think svg/raster is really painful, though I don't have any simple proposition to improve the situation.

I didn't have a good solution either, which was why I pushed it to make the caller responsibility for setting it!

**Comment by nirvn** [comment]
I've tried the fix on my machine, good to go.

**Comment by 3nids** [comment]
The test is a bit difficult to write in current situation. I'll still try to bring something uding the existing layout test.

**Pull request merged**